### ISSUE TITLE
ClassStructure sniff stuck the whole analyze process

### ISSUE THREAD

**Issue by lptn** [slevomat/coding-standard#1191]
It usually takes about 90 secs to scan and analize code, but stuck at some point:
```
Processing PublicationBasedDistinction.php [PHP => 500 tokens in 65 lines]... DONE in 32ms (0 fixable violations)
Processing DistinctionsCollection.php [PHP => 365 tokens in 54 lines]... 
```

the same codebase, -vvv mode, latest output:
```
[PHP => 365 tokens in 54 lines]... 
        *** START TOKEN PROCESSING ***
                Process token 0: T_OPEN_TAG => <?php·
                        Processing IxDFCodingStandard\Sniffs\Files\NonExistingTemplateSniff... DONE in 0.0003 seconds
                        Processing PHP_CodeSniffer\Standards\PSR12\Sniffs\Files\OpenTagSniff... DONE in 0.0001 seconds
                        Processing PHP_CodeSniffer\Standards\PSR1\Sniffs\Files\SideEffectsSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\PHP\DisallowAlternativePHPTagsSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\PHP\DisallowShortOpenTagSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\Files\LineEndingsSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\PSR2\Sniffs\Files\EndFileNewlineSniff... DONE in 0.0002 seconds
                        Processing PHP_CodeSniffer\Standards\PSR2\Sniffs\Files\ClosingTagSniff... DONE in 0.0001 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\Files\LineLengthSniff... DONE in 0.0002 seconds
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\WhiteSpace\SuperfluousWhitespaceSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\WhiteSpace\ScopeIndentSniff... DONE in 0.0008 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\WhiteSpace\DisallowTabIndentSniff... DONE in 0.0002 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\Classes\DuplicateClassNameSniff... DONE in 0.0004 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\PHP\CharacterBeforePHPOpeningTagSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\VersionControl\GitMergeConflictSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Zend\Sniffs\Debug\CodeAnalyzerSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Exceptions\ReferenceThrowableOnlySniff... DONE in 0.001 seconds
                        Processing SlevomatCodingStandard\Sniffs\Namespaces\UselessAliasSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Namespaces\AlphabeticallySortedUsesSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Namespaces\FullyQualifiedGlobalConstantsSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Namespaces\UnusedUsesSniff... DONE in 0.0005 seconds
                        Processing SlevomatCodingStandard\Sniffs\Namespaces\UseSpacingSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Files\LineLengthSniff... DONE in 0.0004 seconds
                        Processing SlevomatCodingStandard\Sniffs\TypeHints\DeclareStrictTypesSniff... DONE in 0 seconds
                Process token 1: T_DECLARE => declare
                        Processing PHP_CodeSniffer\Standards\PSR12\Sniffs\Files\DeclareStatementSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\PHP\LowerCaseKeywordSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\WhiteSpace\ScopeClosingBraceSniff... DONE in 0 seconds
                Process token 2: T_OPEN_PARENTHESIS => (
                        Processing SlevomatCodingStandard\Sniffs\PHP\UselessParenthesesSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Functions\RequireTrailingCommaInCallSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Functions\DisallowTrailingCommaInCallSniff... DONE in 0 seconds
                Process token 3: T_STRING => strict_types
                        Processing IxDFCodingStandard\Sniffs\Functions\ForbiddenFunctionsSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\NamingConventions\UpperCaseConstantNameSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\Functions\FunctionCallArgumentSpacingSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\PSR2\Sniffs\Methods\FunctionCallSignatureSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\Functions\CallTimePassByReferenceSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\PHP\ForbiddenFunctionsSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\PHP\DeprecatedFunctionsSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Classes\ModernClassNameReferenceSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\PHP\DisallowDirectMagicInvokeCallSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\PHP\OptimizedFunctionsWithoutUnpackingSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Functions\StrictCallSniff... DONE in 0.0018 seconds
                        Processing SlevomatCodingStandard\Sniffs\Functions\RequireSingleLineCallSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Functions\RequireMultiLineCallSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\PHP\LowercasePHPFunctionsSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\PHP\DiscouragedFunctionsSniff... DONE in 0 seconds
                Process token 4: T_EQUAL => =
                        Processing SlevomatCodingStandard\Sniffs\Operators\RequireCombinedAssignmentOperatorSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Variables\DuplicateAssignmentToVariableSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\ControlStructures\RequireNullCoalesceEqualOperatorSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\WhiteSpace\OperatorSpacingSniff... DONE in 0 seconds
                Process token 5: T_LNUMBER => 1
                        Processing SlevomatCodingStandard\Sniffs\Numbers\RequireNumericLiteralSeparatorSniff... DONE in 0.0018 seconds
                Process token 6: T_CLOSE_PARENTHESIS => )
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\Functions\FunctionCallArgumentSpacingSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\PSR2\Sniffs\Methods\FunctionCallSignatureSniff... DONE in 0 seconds
                Process token 7: T_SEMICOLON => ;
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\Formatting\DisallowMultipleStatementsSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\PHP\UselessSemicolonSniff... DONE in 0 seconds
                Process token 8: T_WHITESPACE => \n
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\WhiteSpace\SuperfluousWhitespaceSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Whitespaces\DuplicateSpacesSniff... DONE in 0 seconds
                Process token 9: T_WHITESPACE => \n
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\WhiteSpace\SuperfluousWhitespaceSniff... DONE in 0.0018 seconds
                        Processing SlevomatCodingStandard\Sniffs\Whitespaces\DuplicateSpacesSniff... DONE in 0 seconds
                Process token 10: T_NAMESPACE => namespace
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\PHP\LowerCaseKeywordSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\WhiteSpace\ScopeClosingBraceSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Namespaces\NamespaceSpacingSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Namespaces\NamespaceDeclarationSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Namespaces\RequireOneNamespaceInFileSniff... DONE in 0 seconds
                Process token 11: T_WHITESPACE => ·
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\WhiteSpace\SuperfluousWhitespaceSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Whitespaces\DuplicateSpacesSniff... DONE in 0 seconds
                Process token 12: T_STRING => App
                        Processing IxDFCodingStandard\Sniffs\Functions\ForbiddenFunctionsSniff... DONE in 0.0016 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\NamingConventions\UpperCaseConstantNameSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\Functions\FunctionCallArgumentSpacingSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\PSR2\Sniffs\Methods\FunctionCallSignatureSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\Functions\CallTimePassByReferenceSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\PHP\ForbiddenFunctionsSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\PHP\DeprecatedFunctionsSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Classes\ModernClassNameReferenceSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\PHP\DisallowDirectMagicInvokeCallSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\PHP\OptimizedFunctionsWithoutUnpackingSniff... DONE in 0.0019 seconds
                        Processing SlevomatCodingStandard\Sniffs\Functions\StrictCallSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Functions\RequireSingleLineCallSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Functions\RequireMultiLineCallSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\PHP\LowercasePHPFunctionsSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\PHP\DiscouragedFunctionsSniff... DONE in 0 seconds
                Process token 13: T_NS_SEPARATOR => \
                Process token 14: T_STRING => Models
                        Processing IxDFCodingStandard\Sniffs\Functions\ForbiddenFunctionsSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\NamingConventions\UpperCaseConstantNameSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\Functions\FunctionCallArgumentSpacingSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\PSR2\Sniffs\Methods\FunctionCallSignatureSniff... DONE in 0.0017 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\Functions\CallTimePassByReferenceSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\PHP\ForbiddenFunctionsSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\PHP\DeprecatedFunctionsSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Classes\ModernClassNameReferenceSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\PHP\DisallowDirectMagicInvokeCallSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\PHP\OptimizedFunctionsWithoutUnpackingSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Functions\StrictCallSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Functions\RequireSingleLineCallSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Functions\RequireMultiLineCallSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\PHP\LowercasePHPFunctionsSniff... DONE in 0.0017 seconds
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\PHP\DiscouragedFunctionsSniff... DONE in 0 seconds
                Process token 15: T_NS_SEPARATOR => \
                Process token 16: T_STRING => Member
                        Processing IxDFCodingStandard\Sniffs\Functions\ForbiddenFunctionsSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\NamingConventions\UpperCaseConstantNameSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\Functions\FunctionCallArgumentSpacingSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\PSR2\Sniffs\Methods\FunctionCallSignatureSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\Functions\CallTimePassByReferenceSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\PHP\ForbiddenFunctionsSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\PHP\DeprecatedFunctionsSniff... DONE in 0.0022 seconds
                        Processing SlevomatCodingStandard\Sniffs\Classes\ModernClassNameReferenceSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\PHP\DisallowDirectMagicInvokeCallSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\PHP\OptimizedFunctionsWithoutUnpackingSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Functions\StrictCallSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Functions\RequireSingleLineCallSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Functions\RequireMultiLineCallSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\PHP\LowercasePHPFunctionsSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\PHP\DiscouragedFunctionsSniff... DONE in 0 seconds
                Process token 17: T_NS_SEPARATOR => \
                Process token 18: T_STRING => Distinction
                        Processing IxDFCodingStandard\Sniffs\Functions\ForbiddenFunctionsSniff... DONE in 0.0018 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\NamingConventions\UpperCaseConstantNameSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\Functions\FunctionCallArgumentSpacingSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\PSR2\Sniffs\Methods\FunctionCallSignatureSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\Functions\CallTimePassByReferenceSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\PHP\ForbiddenFunctionsSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\PHP\DeprecatedFunctionsSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Classes\ModernClassNameReferenceSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\PHP\DisallowDirectMagicInvokeCallSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\PHP\OptimizedFunctionsWithoutUnpackingSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Functions\StrictCallSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Functions\RequireSingleLineCallSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Functions\RequireMultiLineCallSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\PHP\LowercasePHPFunctionsSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\PHP\DiscouragedFunctionsSniff... DONE in 0 seconds
                Process token 19: T_SEMICOLON => ;
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\Formatting\DisallowMultipleStatementsSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\PHP\UselessSemicolonSniff... DONE in 0 seconds
                Process token 20: T_WHITESPACE => \n
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\WhiteSpace\SuperfluousWhitespaceSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Whitespaces\DuplicateSpacesSniff... DONE in 0 seconds
                Process token 21: T_WHITESPACE => \n
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\WhiteSpace\SuperfluousWhitespaceSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Whitespaces\DuplicateSpacesSniff... DONE in 0 seconds
                Process token 22: T_USE => use
                        Processing PHP_CodeSniffer\Standards\PSR12\Sniffs\Files\ImportStatementSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\PHP\LowerCaseKeywordSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\WhiteSpace\ScopeClosingBraceSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Namespaces\UseDoesNotStartWithBackslashSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Namespaces\MultipleUsesPerLineSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Namespaces\UseFromSameNamespaceSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Functions\UnusedInheritedVariablePassedToClosureSniff... DONE in 0.0017 seconds
                Process token 23: T_WHITESPACE => ·
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\WhiteSpace\SuperfluousWhitespaceSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Whitespaces\DuplicateSpacesSniff... DONE in 0 seconds
                Process token 24: T_STRING => Illuminate
                        Processing IxDFCodingStandard\Sniffs\Functions\ForbiddenFunctionsSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\NamingConventions\UpperCaseConstantNameSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\Functions\FunctionCallArgumentSpacingSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\PSR2\Sniffs\Methods\FunctionCallSignatureSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\Functions\CallTimePassByReferenceSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\PHP\ForbiddenFunctionsSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\PHP\DeprecatedFunctionsSniff... DONE in 0.0017 seconds
                        Processing SlevomatCodingStandard\Sniffs\Classes\ModernClassNameReferenceSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\PHP\DisallowDirectMagicInvokeCallSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\PHP\OptimizedFunctionsWithoutUnpackingSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Functions\StrictCallSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Functions\RequireSingleLineCallSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Functions\RequireMultiLineCallSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\PHP\LowercasePHPFunctionsSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\PHP\DiscouragedFunctionsSniff... DONE in 0 seconds
                Process token 25: T_NS_SEPARATOR => \
                Process token 26: T_STRING => Support
                        Processing IxDFCodingStandard\Sniffs\Functions\ForbiddenFunctionsSniff... DONE in 0.0017 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\NamingConventions\UpperCaseConstantNameSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\Functions\FunctionCallArgumentSpacingSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\PSR2\Sniffs\Methods\FunctionCallSignatureSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\Functions\CallTimePassByReferenceSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\PHP\ForbiddenFunctionsSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\PHP\DeprecatedFunctionsSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Classes\ModernClassNameReferenceSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\PHP\DisallowDirectMagicInvokeCallSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\PHP\OptimizedFunctionsWithoutUnpackingSniff... DONE in 0.0016 seconds
                        Processing SlevomatCodingStandard\Sniffs\Functions\StrictCallSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Functions\RequireSingleLineCallSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Functions\RequireMultiLineCallSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\PHP\LowercasePHPFunctionsSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\PHP\DiscouragedFunctionsSniff... DONE in 0 seconds
                Process token 27: T_NS_SEPARATOR => \
                Process token 28: T_STRING => Collection
                        Processing IxDFCodingStandard\Sniffs\Functions\ForbiddenFunctionsSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\NamingConventions\UpperCaseConstantNameSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\Functions\FunctionCallArgumentSpacingSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\PSR2\Sniffs\Methods\FunctionCallSignatureSniff... DONE in 0.0015 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\Functions\CallTimePassByReferenceSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\PHP\ForbiddenFunctionsSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\PHP\DeprecatedFunctionsSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Classes\ModernClassNameReferenceSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\PHP\DisallowDirectMagicInvokeCallSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\PHP\OptimizedFunctionsWithoutUnpackingSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Functions\StrictCallSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Functions\RequireSingleLineCallSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Functions\RequireMultiLineCallSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\PHP\LowercasePHPFunctionsSniff... DONE in 0.0017 seconds
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\PHP\DiscouragedFunctionsSniff... DONE in 0 seconds
                Process token 29: T_SEMICOLON => ;
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\Formatting\DisallowMultipleStatementsSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\PHP\UselessSemicolonSniff... DONE in 0 seconds
                Process token 30: T_WHITESPACE => \n
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\WhiteSpace\SuperfluousWhitespaceSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Whitespaces\DuplicateSpacesSniff... DONE in 0 seconds
                Process token 31: T_WHITESPACE => \n
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\WhiteSpace\SuperfluousWhitespaceSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Whitespaces\DuplicateSpacesSniff... DONE in 0 seconds
                Process token 32: T_DOC_COMMENT_OPEN_TAG => /**
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\Commenting\FixmeSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Namespaces\FullyQualifiedClassNameInAnnotationSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\TypeHints\LongTypeHintsSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\TypeHints\NullTypeHintOnLastPositionSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\TypeHints\DisallowMixedTypeHintSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Commenting\RequireOneLineDocCommentSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Commenting\ForbiddenAnnotationsSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Commenting\InlineDocCommentDeclarationSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Commenting\RequireOneLinePropertyDocCommentSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Commenting\DocCommentSpacingSniff... DONE in 0.0018 seconds
                        Processing SlevomatCodingStandard\Sniffs\Commenting\ForbiddenCommentsSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Commenting\DeprecatedAnnotationDeclarationSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Commenting\EmptyCommentSniff... DONE in 0.0001 seconds
                Process token 33: T_DOC_COMMENT_WHITESPACE => \n
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\WhiteSpace\SuperfluousWhitespaceSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\Commenting\FixmeSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Whitespaces\DuplicateSpacesSniff... DONE in 0 seconds
                Process token 34: T_DOC_COMMENT_WHITESPACE => ·
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\WhiteSpace\SuperfluousWhitespaceSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\Commenting\FixmeSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Whitespaces\DuplicateSpacesSniff... DONE in 0.0016 seconds
                Process token 35: T_DOC_COMMENT_STAR => *
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\Commenting\FixmeSniff... DONE in 0 seconds
                Process token 36: T_DOC_COMMENT_WHITESPACE => ·
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\WhiteSpace\SuperfluousWhitespaceSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\Commenting\FixmeSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Whitespaces\DuplicateSpacesSniff... DONE in 0 seconds
                Process token 37: T_DOC_COMMENT_STRING => Represents·a·collection·of·all·known·distinctions
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\Commenting\FixmeSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Whitespaces\DuplicateSpacesSniff... DONE in 0 seconds
                Process token 38: T_DOC_COMMENT_WHITESPACE => \n
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\WhiteSpace\SuperfluousWhitespaceSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\Commenting\FixmeSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Whitespaces\DuplicateSpacesSniff... DONE in 0 seconds
                Process token 39: T_DOC_COMMENT_WHITESPACE => ·
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\WhiteSpace\SuperfluousWhitespaceSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\Commenting\FixmeSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Whitespaces\DuplicateSpacesSniff... DONE in 0 seconds
                Process token 40: T_DOC_COMMENT_CLOSE_TAG => */
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\Commenting\FixmeSniff... DONE in 0 seconds
                Process token 41: T_WHITESPACE => \n
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\WhiteSpace\SuperfluousWhitespaceSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Whitespaces\DuplicateSpacesSniff... DONE in 0.0024 seconds
                Process token 42: T_FINAL => final
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\PHP\LowerCaseKeywordSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\Functions\LowercaseFunctionKeywordsSniff... DONE in 0 seconds
                Process token 43: T_WHITESPACE => ·
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\WhiteSpace\SuperfluousWhitespaceSniff... DONE in 0 seconds
                        Processing SlevomatCodingStandard\Sniffs\Whitespaces\DuplicateSpacesSniff... DONE in 0 seconds
                Process token 44: T_CLASS => class
                        Processing PHP_CodeSniffer\Standards\PSR12\Sniffs\Classes\ClosingBraceSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\PSR1\Sniffs\Classes\ClassDeclarationSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\Classes\ValidClassNameSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\PHP\LowerCaseKeywordSniff... DONE in 0.0018 seconds
                        Processing PHP_CodeSniffer\Standards\PSR2\Sniffs\Classes\ClassDeclarationSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Squiz\Sniffs\WhiteSpace\ScopeClosingBraceSniff... DONE in 0 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\CodeAnalysis\UnnecessaryFinalModifierSniff... DONE in 0.0001 seconds
                        Processing PHP_CodeSniffer\Standards\Generic\Sniffs\Files\OneObjectStructurePerFileSniff... DONE in 0.0001 seconds
                        Processing SlevomatCodingStandard\Sniffs\Classes\ClassMemberSpacingSniff... DONE in 0.0002 seconds
                        Processing SlevomatCodingStandard\Sniffs\Classes\ClassStructureSniff... 


````
... and then stuck forever

There are not any messages or errors related to memory, but I assume somehow connected to system resources: on a small codebase it works fine, on a big codebase it stucks

PHP: 8.0.3
slevomat/coding-standard: master branch (ref 3ee23334c19e388a33068a88b25abaef61a10326)

**Comment by kukulich** [comment]
@lptn You have to find the problematic sniff and also your code (file) that caused this.

**Comment by lptn** [comment]
It was `SlevomatCodingStandard.Classes.ClassStructure`

code (simplified and useless):
```php
<?php

namespace App;

use Illuminate\Support\Collection;

class CustomCollection extends Collection
{
    public function findBySlug(string $slug)
    {
        return (new Collection())
            ->first(static fn ($distinction) => $distinction->getSlug() === $slug);
    }
}

```

**Comment by kukulich** [comment]
Thank you for your report.

Fixed in https://github.com/slevomat/coding-standard/commit/8e049b5df74441fd6fa920ba986de7941d416892
